pkgs/tools/misc/ugs/default.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv
+, lib
+, copyDesktopItems
+, makeDesktopItem
+, makeWrapper
+, jre
+, fetchzip
+, bash
+}:
+let
+  desktopItem = makeDesktopItem {
+    name = "ugs";
+    exec = "ugs";
+    comment = "A cross-platform G-Code sender for GRBL, Smoothieware, TinyG and G2core.";
+    desktopName = "Universal-G-Code-Sender";
+    categories = [ "Game" ];
+  };
+
+in
+stdenv.mkDerivation rec {
```

Comment:
How about using directly `finalAttrs` pattern here?

Reply:
Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Once again these are non-blocking comments, if you don't want to implement them it's perfectly fine as well !

It's just that I guess, at some point, we will switch to this pattern in `nixpkgs`.

Reply:
I think this specific case is a case where `finalAttrs` is not necessary, since it's only used to set the version number, which when overridden via overrideAttrs will always be changed by the user. So in this case, `finalAttrs` does not add anything, so I won't implement it.